slay/truss_adapter/deploy.py

```diff
@@ -0,0 +1,325 @@
+            for name in names
+            if os.path.isfile(os.path.join(path, name))
+            and not name.endswith(".py")
+            or definitions.GENERATED_CODE_DIR in name
+        ]
+
+    shutil.copytree(root_dir, dest_dir, ignore=python_files_only, dirs_exist_ok=True)
+
+
+def _make_truss_config(
+    truss_dir: pathlib.Path,
+    slay_config: definitions.Config,
+    stub_cls_to_url: Mapping[str, str],
+    model_name: str,
+) -> truss_config.TrussConfig:
+    """Generate a truss config for a processor."""
+    config = truss_config.TrussConfig()
+    config.model_name = model_name
+    config.model_class_filename = _MODEL_CLASS_FILENAME
+    config.model_class_name = _MODEL_CLASS_NAME
+    # Compute.
+    compute = slay_config.get_compute_spec()
+    config.resources.cpu = compute.cpu
+    config.resources.accelerator = truss_config.AcceleratorSpec.from_str(compute.gpu)
+    config.resources.use_gpu = bool(compute.gpu)
+    # Image.
+    image = slay_config.get_image_spec()
+    config.base_image = truss_config.BaseImage(image=image.base_image)
+    pip_requirements: list[str] = []
+    if image.pip_requirements_file:
+        pip_requirements.extend(
+            req
+            for req in pathlib.Path(image.pip_requirements_file)
+            .read_text()
+            .splitlines()
+            if not req.strip().startswith("#")
+        )
+    pip_requirements.extend(image.pip_requirements)
+    # `pip_requirements` will add server requirements which give version conflicts.
+    # config.requirements = pip_requirements
+    pip_requirements_file_path = truss_dir / _REQUIREMENTS_FILENAME
+    pip_requirements_file_path.write_text("\n".join(pip_requirements))
+    # TODO: apparently absolute paths don't work with remote build (but work in local).
+    config.requirements_file = _REQUIREMENTS_FILENAME  # str(pip_requirements_file_path)
+    config.system_packages = image.apt_requirements
+    # Assets.
+    assets = slay_config.get_asset_spec()
+    config.secrets = assets.secrets
+    if definitions.BASTEN_APY_SECRET_NAME not in config.secrets:
+        config.secrets[definitions.BASTEN_APY_SECRET_NAME] = "***"
+    else:
+        logging.info(
+            f"Workflows automatically add {definitions.BASTEN_APY_SECRET_NAME} "
+            "to secrets - no need to manually add it."
+        )
+    config.model_cache.models = assets.cached
+    # Metadata.
+    slay_metadata: definitions.TrussMetadata = definitions.TrussMetadata(
+        user_config=slay_config.user_config,
+        stub_cls_to_url=stub_cls_to_url,
+    )
+    config.model_metadata[definitions.TRUSS_CONFIG_SLAY_KEY] = slay_metadata.dict()
+    return config
+
+
+def make_truss(
+    processor_dir: pathlib.Path,
+    workflow_root: pathlib.Path,
+    slay_config: definitions.Config,
+    model_name: str,
+    stub_cls_to_url: Mapping[str, str],
+    maybe_stub_file: Optional[pathlib.Path],
+) -> pathlib.Path:
+    truss_dir = processor_dir / _TRUSS_DIR
```

Comment:
I think it might be easier to make truss init just take a config, string text for model class, but this works too. Just seems to be duplicating code that really should live in truss.

Reply:
Not sure how this could live in truss, `_make_truss_config` is the "adapter layer" between truss config and the slay APIs and as such specific to slay.

I can revisit this point, if you let me know which functionality exactly you'd like to have moved into truss.